server/service/src/sync/integrate_document.rs

```diff
@@ -10,15 +12,31 @@ use crate::programs::{
     patient::{patient_schema::SchemaPatient, patient_updated::update_patient_row},
     program_enrolment::program_enrolment_updated::update_program_enrolment_row,
     program_enrolment::program_schema::SchemaProgramEnrolment,
+    update_program_document::update_program_events,
 };
 
 pub(crate) fn sync_upsert_document(
     con: &StorageConnection,
     document: &Document,
 ) -> Result<(), RepositoryError> {
-    // TODO comment why only insert here
+    // Fetch current document by name to check if the new document is the latest in the DB
+    let current_doc = DocumentRepository::new(con)
```

Comment:
One of the problems we've identified with existing mSupply code base was "same logic is repeated in different places, slight variations". I think the main reasons why it'se problematic are:
* Hard to know where this logic needs to be adjusted if changed
* More code to digest and validate

It's great to see the tests being added, it kind of helps with above two points, but I still think we should do [something like this](https://github.com/openmsupply/open-msupply/compare/1978-update-program-events-during-sync...1978-update-program-events-during-sync(Consolidate-latest-document-helper)) (btw also added a comment that DocumentRepository will always grab the latest document for a document name).

Reply:
thanks got a bit lazy. With a previous version of `is_latest_doc` there were some issue depending on if you use it before or after you inserting the current document and so avoided touching it... But it should work as it is now! I tweaked the doc comment a bit more to make it clearer what the method exactly does...

Reply:
thank you